src/goto-programs/goto_inline_class.cpp

```diff
@@ -0,0 +1,1297 @@
+/*******************************************************************\
+
+Module: Function Inlining
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+//#define DEBUG
+
+#include <iostream>
+#include <cassert>
+
+#include <util/prefix.h>
+#include <util/cprover_prefix.h>
+#include <util/base_type.h>
+#include <util/std_code.h>
+#include <util/std_expr.h>
+#include <util/expr_util.h>
+
+#include "remove_skip.h"
+#include "goto_inline.h"
+#include "goto_inline_class.h"
+
+/*******************************************************************\
+
+Function: goto_inlinet::parameter_assignments
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void goto_inlinet::parameter_assignments(
+  const goto_programt::targett target,
+  const irep_idt &function_name, // name of called function
+  const code_typet &code_type, // type of called function
+  const exprt::operandst &arguments, // arguments of call
+  goto_programt &dest)
+{
+  assert(is_call(target));
+  assert(dest.empty());
+
+  const source_locationt &source_location=target->source_location;
+
+  // iterates over the operands
+  exprt::operandst::const_iterator it1=arguments.begin();
+
+  const code_typet::parameterst &parameter_types=
+    code_type.parameters();
+  
+  // iterates over the types of the parameters
+  for(code_typet::parameterst::const_iterator
+      it2=parameter_types.begin();
+      it2!=parameter_types.end();
+      it2++)
```

Comment:
My understanding is that this is existing code. In future, this should be moved to a ranged for.